### PR TITLE
Supporting Special Characters in Unicode for Search Functions

### PR DESCRIPTION
**Pull request by whomingbird**
https://github.com/seek4science/seek/issues/1965

### CODE REVIEW

**Comment by whomingbird** [comment]
> is this intended for 1.16.0 ?
yes.

File: solr/seek/conf/mapping-hyphen.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Mapping of Unicode hyphen to ASCII hyphen
+"\u2010" => "-"
+"\u2011" => "-"
+"\u2012" => "-"
+"\u2013" => "-"
+"\u2014" => "-"
+"\u2015" => "-"
```
Comment (stuzart) [inline solr/seek/conf/mapping-hyphen.txt:7]: I don't this file is needed, as they are already included in teh FoldToASCII

File: test/functional/search_controller_test.rb [hunk 2]
```diff
@@ -155,6 +155,58 @@ class SearchControllerTest < ActionController::TestCase
     assert_select 'a[href="#test-123"]', count: 0
   end
 
+
+  test 'hyphen and unicode hyphen are treated equally in search' do
```
Comment (stuzart) [inline test/functional/search_controller_test.rb:159]: you realise this isn't testing the search, as it is stubbed out ?
but still useful I think for checking the flash notice text contains the original text.

**Review by stuzart — CHANGES_REQUESTED** [review]

**Comment by stuzart** [comment]
The docker container at https://github.com/FAIRdom/solr-seek-docker also needs updating. If you update this too, on a branch, we can give a test on the test server before merging

File: solr/seek/conf/mapping-hyphen.txt (hunk 1 above)
Comment (whomingbird) [reply solr/seek/conf/mapping-hyphen.txt:7]: i just realized that

**Review by stuzart — APPROVED** [review]

**Pull request merged**